File: 0001. Two Sum/solution.go
package solution

func twoSum(nums []int, target int) []int {
	hash := make(map[int]int)
	for i, val := range nums {
		if v, ok := hash[target-val]; ok {
			return []int{i, v}
		}
		// val對應索引值
		hash[val] = i
	}
	return nil
}

// 4 ms, faster than 94.09% of Go online submissions for Two Sum.
// 3.8 MB, less than 49.61% of Go online submissions for Two Sum.

File: 0001. Two Sum/solution.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        vector<int> solve = {};
        for(int index = 0; index < nums.size(); ++index) {
            if(hash.find(target-nums[index]) != hash.end()) {
                solve.push_back(index);
                solve.push_back(hash[target-nums[index]]);
                return solve;
            }
            hash[nums[index]] = index;
        }
        return solve;
    }
};